src/services.ts

```diff
@@ -30,33 +32,161 @@ export class ServiceCompletionProvider implements vscode.CompletionItemProvider
+                            params.push(paramText)
+                        }
+
+                        const completionItem = new vscode.CompletionItem(
+                            member.Name,
+                            vscode.CompletionItemKind.Event,
+                        )
+                        completionItem.detail = `(event) ${service.Name}.${member.Name}(${params.join(", ")})`
+                        completionItem.documentation = new vscode.MarkdownString(`[Developer Reference](https://developer.roblox.com/en-us/api-reference/property/${service.Name}/${member.Name})`)
+
+                        completionItems.push(completionItem)
+                        break
+                    }
+                    case "Property": {
+                        const completionItem = new vscode.CompletionItem(
+                            member.Name,
+                            vscode.CompletionItemKind.Field,
+                        )
+                        completionItem.detail = `(property) ${service.Name}.${member.Name}: ${member.ValueType ? member.ValueType.Name : "unknown"}`
+                        completionItem.documentation = new vscode.MarkdownString(`[Developer Reference](https://developer.roblox.com/en-us/api-reference/event/${service.Name}/${member.Name})`)
+                        completionItems.push(completionItem)
+                        break
+                    }
+                }
+            }
+        }
+
+        if (inheritMembers) {
+            if (service.Superclass) {
+                const klass = (await this.serviceMembers).get(service.Superclass)
+                if (klass) {
+                    const inheritedMembers = await this.createCompletionItems(klass, operator, true)
+                    for (const completionItem of inheritedMembers) {
+                        if (completionItem.documentation) {
+                            (completionItem.documentation as vscode.MarkdownString).value = `Inherited from ${service.Superclass}\n\n${(completionItem.documentation as vscode.MarkdownString).value}`
+                        }
+                    }
+                    completionItems = completionItems.concat(inheritedMembers)
+                }
+            }
+        }
+
+        return completionItems
+    }
+
+    public async provideCompletionItems(document: vscode.TextDocument, position: vscode.Position) {
         const serviceMatch = document.lineAt(position.line).text.substr(0, position.character).match(/(\w+)([:.]?)\w*$/)
 
         if (serviceMatch !== null) {
             const serviceName = serviceMatch[1]
             const operator = serviceMatch[2]
 
-            const serviceMembers = (await this.serviceMembers).get(serviceName)
+            const service = (await this.serviceMembers).get(serviceName)
 
-            if (serviceMembers !== undefined) {
+            if (service !== undefined && service.Tags && service.Tags.includes("Service")) {
```

Comment:
I generally prefer explicit undefined checks.

```suggestion
             if (service !== undefined && service.Tags !== undefined && service.Tags.includes("Service")) {
```